linkerd/app/integration/src/proxy.rs

```diff
@@ -249,55 +242,68 @@ fn run(proxy: Proxy, mut env: app::config::TestEnv) -> Listening {
         rx = Box::new(rx.select(fut).then(|_| Ok(())));
     }
 
-    let tname = format!("support proxy (test={})", thread_name());
-    ::std::thread::Builder::new()
-        .name(tname)
+    std::thread::Builder::new()
+        .name(format!("{}:proxy", thread_name()))
         .spawn(move || {
             tracing::dispatcher::with_default(&trace, || {
-                let _c = controller;
-                let _i = identity;
-
-                let mock_orig_dst = MockOriginalDst(Arc::new(Mutex::new(mock_orig_dst)));
-                let main = linkerd2_app::Main::new(config, trace_handle, mock_orig_dst.clone());
-
-                {
-                    let mut inner = mock_orig_dst.0.lock().unwrap();
-                    inner.inbound_local_addr = Some(main.inbound_addr());
-                    inner.outbound_local_addr = Some(main.outbound_addr());
-                }
-
-                // slip the running tx into the shutdown future, since the first time
-                // the shutdown future is polled, that means all of the proxy is now
-                // running.
-                let addrs = (
-                    main.control_addr(),
-                    identity_addr,
-                    main.inbound_addr(),
-                    main.outbound_addr(),
-                    main.metrics_addr(),
-                );
-                let mut running = Some((running_tx, addrs));
-                let on_shutdown = future::poll_fn(move || {
-                    if let Some((tx, addrs)) = running.take() {
-                        let _ = tx.send(addrs);
-                    }
-
-                    try_ready!(rx.poll());
-                    Ok(().into())
-                });
-
-                main.run_until(on_shutdown);
+                info_span!("proxy", test = %thread_name()).in_scope(|| {
```

Comment:
<3 <3 <3

Reply:
Note that if we changed this to
```rust
let span = info_span!("proxy", test = %thread_name());
let _guard = span.enter()
```
we wouldn't have to indent the entire thing over another indentation level.